src/Storage/Storage.Management/Common/AzureContextAdapterExtensions.cs

```diff
@@ -0,0 +1,134 @@
+                }
+                catch
+                {
+                    var service = provider.GetStorageService(storageConnectionString, null);
+                    return service.GetCloudStorageAccount();
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Get a CloudStorageAccount client for the storage account represented by this context
+        /// </summary>
+        /// <param name="context">The storage context for the storage account</param>
+        /// <returns>A CloudStorageAccount client for storage data plane tasks</returns>
+        public static CloudStorageAccount GetCloudStorageAccount(this IStorageContext context)
+        {
+            CloudStorageAccount result = null;
+            CloudStorageAccount.TryParse(context.ConnectionString, out result);
+            return result;
+        }
+
+        /// <summary>
+        /// Get a CloudStorageAccount client for the storage account represented by this service
+        /// </summary>
+        /// <param name="service">The storage service</param>
+        /// <returns>A CloudStorageAccount client for storage data plane tasks</returns>
+        public static CloudStorageAccount GetCloudStorageAccount( this IStorageService service)
+        {
+            return new CloudStorageAccount(new StorageCredentials(service.Name, service.AuthenticationKeys.First()),
+                new StorageUri(service.BlobEndpoint), new StorageUri(service.QueueEndpoint), 
+                new StorageUri(service.TableEndpoint), new StorageUri(service.FileEndpoint));
+        }
+
+        /// <summary>
+        /// Get a storage context client for the storage account represented by this service
+        /// </summary>
+        /// <param name="service">The storage service</param>
+        /// <returns>A CloudStorageAccount client for storage data plane tasks</returns>
+        public static IStorageContext GetStorageContext(this IStorageService service)
+        {
+            return new AzureStorageContext(new CloudStorageAccount(new StorageCredentials(service.Name, service.AuthenticationKeys.First()),
+                new StorageUri(service.BlobEndpoint), new StorageUri(service.QueueEndpoint),
+                new StorageUri(service.TableEndpoint), new StorageUri(service.FileEndpoint)));
+        }
+
+
+        /// <summary>
+        /// Get a CloudStorageAccount client for the given storage service using the given storage service provider
+        /// </summary>
+        /// <param name="provider">The storage service provider to retrieve storage service details</param>
+        /// <param name="accountName">The storage accoutn name</param>
+        /// <returns>A CloudStorageAccount client for storage data plane tasks</returns>
+        public static CloudStorageAccount GetCloudStorageAccount(this IStorageServiceProvider provider, string accountName, string resourceGroupName = null)
+        {
+#if DEBUG
```

Comment:
Why can't you use the provided StorageServiceProvider in debug mode?

Reply:
actually, this is just copied from XSCL wrapper, with slightly change on namespace. 
This part code is not written by me. (@markcowl , Do you write it?)